itl/command-plugins.conf

```diff
@@ -1335,6 +1339,14 @@ object CheckCommand "ssh" {
 			set_if = "$ssh_ipv6$"
 			description = "Use IPv6 connection"
 		}
+		"-r" = {
+			value = "$ssh_remote_version$"
+			description = "Alert if string doesn't match expected server version (ex: OpenSSH_3.9p1)"
+		}
+		"-P" = {
+			value = "$ssh_remote_protocol$"
+			description = "Alert if protocol doesn't match expected protocol version (ex: 2.0)"
+		}
```

Comment:
Please use `set_if` instead of `value` here as well, otherwise Icinga will pass these args with an empty string and will result into a [CRITICAL check](https://github.com/monitoring-plugins/monitoring-plugins/blob/master/plugins/check_ssh.c#L330-L345) even though the user didn't explicitly set them to anything.

```c
#include <stdio.h>
#include <string.h>

int main() {
  if ("" && strcmp("", "OpenSSH_3")) {
    printf("REMOTE VERSION MISMATCH\n");
  }

  return 0;
}

./a.out
REMOTE VERSION MISMATCH
```

Reply:
Since when are parameters set with empty values, as long as I don't set the value of ssh_remote_protocol/ssh_remote_version? Oo

Reply:
They are not! It was my mistake to think that they behave like that, but if you don't explicitly set them to empty values, Icinga 2 won't pass them to the plugin.